### ISSUE TITLE
fastlane fails to read from keychain and always prompts for password

### ISSUE THREAD

**Issue by dansimau** [fastlane/fastlane#20952]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

I added credentials using:

```
fastlane fastlane-credentials add --username me@example.com
```

Then I execute my lane, which requires auth:

```
fastlane ios beta
```

I get the following:

```
-------------------------------------------------------------------------------------
Please provide your Apple Developer Program account credentials
The login information you enter will be stored in your macOS Keychain
You can also pass the password using the `FASTLANE_PASSWORD` environment variable
See more information about it on GitHub: https://github.com/fastlane/fastlane/tree/master/credentials_manager
-------------------------------------------------------------------------------------
Password (for me@example.com):
```

No matter how many times I type my password, it prompts me every single time.

I debugged this and found the issue on the following line:
https://github.com/fastlane-community/security/blob/f6c295b/lib/security/password.rb#L35

The regexp for parsing the parsing from `security` output is: `/^password: "(.+)"/` but this does not match the output from this command on my system.

When I run `security find-internet-password -g -s deliver.me@example.com` I see the following output:

```
keychain: "/Users/me/Library/Keychains/login.keychain-db"
version: 512
class: "inet"
attributes:
    0x00000007 <blob>="deliver.me@example.com"
    0x00000008 <blob>=<NULL>
    "acct"<blob>="me@example.com"
    "atyp"<blob>="dflt"
    "cdat"<timedate>=0x32303232313232373137303830315A00  "20221227170801Z\000"
    "crtr"<uint32>=<NULL>
    "cusi"<sint32>=<NULL>
    "desc"<blob>=<NULL>
    "icmt"<blob>=<NULL>
    "invi"<sint32>=<NULL>
    "mdat"<timedate>=0x32303232313232373137303830315A00  "20221227170801Z\000"
    "nega"<sint32>=<NULL>
    "path"<blob>=<NULL>
    "port"<uint32>=0x00000000
    "prot"<blob>=<NULL>
    "ptcl"<uint32>=0x00000000
    "scrp"<sint32>=<NULL>
    "sdmn"<blob>=<NULL>
    "srvr"<blob>="deliver.me@example.com"
    "type"<uint32>=<NULL>
password: 0x6d7970617373776f7264  "mypassword"
```

Note the line containing the password shows it in both hex and as a string, but the regexp `/^password: "(.+)"/` does not account for the hex string, therefore the password is set to `nil`.

I'm not able to verify but I'm guessing the output of the security command may have changed?

My version of macOS is: Monterey 12.6 (21G115).

**Comment by dansimau** [comment]
Okay it seems this is fixed in: https://github.com/fastlane-community/security/pull/10 (0.1.5)

However fastline is pinned to an older version (0.1.3):
https://github.com/fastlane/fastlane/blob/8e456d6/Gemfile.lock#L33

**Comment by dansimau** [comment]
Fixed by #21818.
